cfme/automate/provisioning_dialogs.py

```diff
@@ -1,138 +1,234 @@
+    HOST_PROVISION = 'Host Provision'
+    VM_MIGRATE = 'VM Migrate'
+    VM_PROVISION = 'VM Provision'
+    SYSTEM_PROVISION = 'Configured System Provision'
+    ALLOWED_TYPES = {HOST_PROVISION, VM_MIGRATE, VM_PROVISION, SYSTEM_PROVISION}
+
+    pretty_attrs = ['name', 'description', 'diag_type', 'content']
+
+    def __init__(self, diag_type, name=None, description=None, content=None, appliance=None):
         Navigatable.__init__(self, appliance=appliance)
         self.name = name
         self.description = description
-        self.type = type
         self.content = content
+        if diag_type in self.ALLOWED_TYPES:
+            self.diag_type = diag_type
+        else:
+            raise TypeError('Type must be one of ProvisioningDialog constants: {}'
+                            .format(self.ALLOWED_TYPES))
 
     def __str__(self):
         return self.name
 
     @property
-    def type_nav(self):
-        return self.type[0]
-
-    @property
-    def type_tree_nav(self):
-        return self.type[1]
-
-    @property
     def exists(self):
         try:
             navigate_to(self, 'Details')
-            return True
         except CandidateNotFound:
             return False
+        return True
 
     def create(self, cancel=False):
-        navigate_to(self, 'Add')
-        fill(self.form, self.__dict__, action=form_buttons.cancel if cancel else form_buttons.add)
-        flash.assert_no_errors()
+        view = navigate_to(self, 'Add')
+        # might not want to use pretty_attrs here, overloading its intended use
+        fill_args = {key: self.__dict__[key] for key in self.pretty_attrs}
+        view.form.fill(fill_args)
+        if cancel:
+            flash_msg = 'Add of new Dialog was cancelled by the user'
+            btn = view.form.cancel
+        else:
+            flash_msg = 'Dialog "{}" was added'.format(self.name)
+            btn = view.form.add
+
+        btn.click()
+        view = self.create_view(ProvDiagAllView if cancel else ProvDiagDetailsView)
+        assert view.is_displayed
+        view.flash.assert_success_message(flash_msg)
+
+    def update(self, updates, cancel=False, reset=False):
+        view = navigate_to(self, 'Edit')
+        view.form.fill(updates)
+        if reset:
+            cancel = True
+            view.form.reset.click()
+            view.flash.assert_message('All changes have been reset')
+        if cancel:
+            flash_msg = 'Edit of Dialog "{}" was cancelled by the user'.format(self.name)
+            btn = view.form.cancel
+        else:
+            flash_msg = ('Dialog "{}" was saved'.format(updates.get('name') or self.name))
+            btn = view.form.save
 
-    def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(self.form, updates, action=form_buttons.save)
-        flash.assert_no_errors()
+        btn.click()
+        view = self.create_view(ProvDiagDetailsView)
+        # self hasn't been updated yet, can't assert is_displayed
```

Comment:
**REQUIRED**
You can just do:
```python
view = self.create_view(ProvDiagDetailsView, override=updates)
```
and after that do assert

Reply:
Initial testing the is_displayed check still failed, debugging a bit because this would be a nice thing to have.

Reply:
Adding a TODO, I'd like to address this at another time with the focus area owner